llvm/test/MC/Disassembler/X86/usermsr-64.txt

```diff
@@ -0,0 +1,26 @@
+# RUN: llvm-mc --disassemble %s -triple=x86_64 | FileCheck %s --check-prefixes=ATT
+# RUN: llvm-mc --disassemble %s -triple=x86_64 --output-asm-variant=1 | FileCheck %s --check-prefixes=INTEL
+
+# ATT:   urdmsr $123, %r9
+# INTEL: urdmsr r9, 123
+0xc4,0xc7,0x7b,0xf8,0xc1,0x7b,0x00,0x00,0x00
+
+# ATT:   urdmsr %r9, %r9
+# INTEL: urdmsr r9, r9
+0xf2,0x45,0x0f,0x38,0xf8,0xc9
+
+# ATT:   urdmsr %r9, %r9
+# INTEL: urdmsr r9, r9
+0xf2,0x4d,0x0f,0x38,0xf8,0xc9
```

Comment:
Why different encodings decode to the same instruction? Should avoid to use the same register?

Reply:
All legacy Instructions with R64 supported could be judged as "software WIG". So this is a WIG instruction, too. Same as below:
```
$ echo "0xf3,0x41,0x0f,0xc7,0xf0" | llvm-mc --disassemble
        .text
        senduipi        %r8
$ echo "0xf3,0x49,0x0f,0xc7,0xf0" | llvm-mc --disassemble
        .text
        senduipi        %r8
```
Will add a comment.

Reply:
3e32e495d8fa56ad3770769a5d4559ea0a41af96